### DISCUSSION TITLE
limited free mints with 1 per wallet limit?

### DISCUSSION THREAD

**Discussion by mpixelz** [hashlips-lab/nft-erc721-collection#457] [Q&A]
Hi,
Thank you for the great repo and tools by hashlips.
I have been following on discord and many of us have this request if someone can guide.
Is there a way to offer 1 free nft to all the new users but limited to a certain number of free nft supply?
For instance, i have collection of 5000, and i want to give out 500 free nfts. Only 1 free nft can be claimed per wallet. but they can still mint paid nft alongside their free nft.

I know workaround would be to have a pre-sale of 0 value and keep an eye on the supply being minted or create a function to update max supply but how can we allow users to mint as many as they want while have the ability to allow them first nft as free?

if you look at discord, many of us are looking to make this sort of contract and reflect that onto the dapp as well.
Thank you.

**Comment by liarco** [comment]
Hi @mpixelz,
At the moment I actively maintain my projects by fixing bugs that people report here on GitHub. For any support for custom implementations/features or previews of future updates I only provide this kind of support to holders of ODC tokens (https://opendevs.io). This helps me focusing my resources on people who support my work and it allows me to work on new tools to bring out in the future with less distractions.

If you are already a holder you can follow the instructions on the website to join our private Discord channel and ask your questions there. We already have some examples and content that might help, but we can also help with more specific stuff.

I hope this helps and I wish you a wonderful day! ☀️
